README.md

```diff
@@ -21,6 +21,22 @@ Note: To be able to take advantage of alternate `define` method name, you will a
 build tooling generates using the alternate.  An example of this is done in the [emberjs-build](https://github.com/emberjs/emberjs-build)
 project in the [babel-enifed-module-formatter plugin](https://github.com/emberjs/emberjs-build/blob/v0.4.2/lib/utils/babel-enifed-module-formatter.js).
 
+## wrapModules
+
+It is possible to hook loader to augment or transform the loaded code.  `wrapModules` is an optional method on the loader that is called as each module is originally loaded.  `wrapModules` must be a function of the form `wrapModules(name, callback)`. The `callback` is the original AMD callback.  The return value of `wrapModules` is then used in subsequent requests for `name`
+
+This functionality is useful for instrumenting code, for instance in code coverage libraries.  
+
+```
+loader.wrapModules = function(name, callback) {
+            if (shouldTransform(name) {
```

Comment:
it’s a fair concern but given that we’ve just arrived here I would defer until such a need arises.  It can be solved in user land.



> On Jan 7, 2016, at 4:16 PM, Stefan Penner <notifications@github.com> wrote:
> 
> In README.md <https://github.com/ember-cli/loader.js/pull/62#discussion_r49126842>:
> 
> > @@ -21,6 +21,22 @@ Note: To be able to take advantage of alternate `define` method name, you will a
> >  build tooling generates using the alternate.  An example of this is done in the [emberjs-build](https://github.com/emberjs/emberjs-build)
> >  project in the [babel-enifed-module-formatter plugin](https://github.com/emberjs/emberjs-build/blob/v0.4.2/lib/utils/babel-enifed-module-formatter.js).
> >  
> > +## wrapModules
> > +
> > +It is possible to hook loader to augment or transform the loaded code.  `wrapModules` is an optional method on the loader that is called as each module is originally loaded.  `wrapModules` must be a function of the form `wrapModules(name, callback)`. The `callback` is the original AMD callback.  The return value of `wrapModules` is then used in subsequent requests for `name`
> > +
> > +This functionality is useful for instrumenting code, for instance in code coverage libraries.  
> > +
> > +```
> > +loader.wrapModules = function(name, callback) {
> > +            if (shouldTransform(name) {
> Should we be concerned that people will have multiple tools that override this method? Should we encourage chaining, or implement chaining behind the scenes?
> 
> Am i overly concerned?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ember-cli/loader.js/pull/62/files#r49126842>.
>